presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQuerySchedulerInterface.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.execution.scheduler;
+
+import com.facebook.presto.execution.BasicStageExecutionStats;
+import com.facebook.presto.execution.StageId;
+import com.facebook.presto.execution.StageInfo;
+import io.airlift.units.Duration;
+
+public interface SqlQuerySchedulerInterface
```

Comment:
hmm... any suggestion on name ? @tdcmeehan ? We can also have
```
interface SqlQueryScheduler
    |-   class LegacySqlQueryScheduler
    |-   class SectionRetriable SqlQueryScheduler
```

Reply:
I prefer the interface be called SqlQueryScheduler.  I didn't name it that because it made rebasing the later changes a gigantic pain.  I can rename it at the end though.

Reply:
@rschlussel : Makes sense, we can have a separate PR or address it in the last commit before merge :)